src/ubx_message.toit

```diff
@@ -0,0 +1,559 @@
+  static ID ::= 0x01
+
+  /** Constructs a dummy acknowledge message. */
+  constructor.private_ cls id:
+    super Message.ACK ID #[cls, id]
+
+  /** The class ID of the acknowleged message. */
+  class_id -> int:
+    return LITTLE_ENDIAN.uint8 payload 0
+
+  /** The message ID  of the acknowleged message. */
+  message_id -> int:
+    return LITTLE_ENDIAN.uint8 payload 1
+
+  id_string_ -> string:
+    return "ACK"
+
+
+/**
+The UBX-ACK-NAK message.
+
+Contains the class ID and message ID of the not acknowledged message.
+*/
+class AckNak extends Message:
+  /** The UBX-ACK-NAK message ID. */
+  static ID ::= 0x02
+
+  /** Constructs a dummy not acknowledge message. */
+  constructor.private_ cls id:
+    super Message.ACK ID #[cls, id]
+
+  /** The class ID of the not acknowlegede message. */
+  class_id -> int:
+    return LITTLE_ENDIAN.uint8 payload 0
+
+  /** The message ID of the not acknowleged message. */
+  message_id -> int:
+    return LITTLE_ENDIAN.uint8 payload 1
+
+  id_string_ -> string:
+    return "NAK"
+
+/**
+The UBX-CFG-MSG message.
+
+Configures the rate at which messages are sent by the receiver.
+*/
+class CfgMsg extends Message:
+  /** The UBX-CFG-MSG message ID. */
+  static ID ::= 0x01
+
+  /**
+  Constructs a coniguration message.
+
+  When sent to the receiver, the message with the given $msg_class and
+    $msg_id will be sent at the given $rate.
+  */
+  constructor.message_rate --msg_class --msg_id --rate:
+    super Message.CFG ID #[msg_class, msg_id, rate]
+
+  id_string_ -> string:
+    return "MSG"
+
+/**
+The UBX-CFG-RST message.
+
+Resets the receiver.
+*/
+class CfgRst extends Message:
+  /** The UBX-CFG-RST message ID. */
+  static ID ::= 0x04
+
+  /**
+  Constructs a reset message.
+
+  The default parameters are a controlled software reset with a cold start.
+  See the description for other parameter options.
+  */
+  constructor --clear_sections=0xFFFF --reset_mode=2:
+    super Message.CFG ID (ByteArray 4)
+    LITTLE_ENDIAN.put_uint16 payload 0 clear_sections
+    LITTLE_ENDIAN.put_uint8  payload 2 reset_mode
+    LITTLE_ENDIAN.put_uint8  payload 3 Message.RESERVED_
+
+  id_string_ -> string:
+    return "RST"
+
+/**
+The UBX-NAV-PVT message.
+
+Navigation, position, velocity, and time solution.
+*/
+class NavPvt extends Message:
+  /** The UBX-NAV-PVT message ID. */
+  static ID ::= 0x07
+
+  /** Unknown GNSS fix. */
+  static FIX_TYPE_UNKNOWN ::= 0
+  /** Dead reckoning only. */
+  static FIX_TYPE_DEAD ::= 1
+  /** 2D fix. */
+  static FIX_TYPE_2D ::= 2
+  /** 3D fix. */
+  static FIX_TYPE_3D ::= 3
+  /** GNSS and dead reckoning. */
+  static FIX_TYPE_GNNS_DEAD ::= 4
+  /** Time only fix. */
+  static FIX_TYPE_TIME_ONLY ::= 5
+
+  /** Constructs a poll UBX-NAV-PVT message. */
+  constructor.poll:
+    super Message.NAV ID #[]
+
+  id_string_ -> string:
+    return "PVT"
+
+  /** Whether the give $message is a UBX-NAV-PVT message. */
+  static is_instance message/Message -> bool:
```

Comment:
Not sure I understand this function.
Why not just `o is NavPvt` ?

Reply:
This checks the class and ID of the actual message rather than the type of the object.

When you receive a message from the receiver, it is just a `Message`. This method let's you test whether the message is actually a `NavPvt`. If you used `o is NavPvt`. then it would not be recognized as a `NavPvt` message (even if it is one).

Reply:
Then maybe add this as a comment.
Also explain how I can go from the generic message to a `NavPvt` message (once I know that it is of that instance).

Generally, we use "instance" for instances of classes in Toit. Maybe rename to `is_of_class` or something similar?

Reply:
https://github.com/toitware/ubx-message/pull/2